lemmy_utils/src/lib.rs

```diff
@@ -68,7 +72,224 @@ impl std::fmt::Display for LemmyError {
+    LemmyError {
+      inner: e.into(),
+      status_code: None,
+    }
+  }
+}
+
+impl From<ToStrError> for LemmyError {
+  fn from(e: ToStrError) -> Self {
+    LemmyError {
+      inner: e.into(),
+      status_code: None,
+    }
+  }
+}
+
+impl From<r2d2::Error> for LemmyError {
+  fn from(e: r2d2::Error) -> Self {
+    LemmyError {
+      inner: e.into(),
+      status_code: None,
+    }
+  }
+}
+
+impl From<BlockingError<LemmyError>> for LemmyError {
+  fn from(e: BlockingError<LemmyError>) -> Self {
+    LemmyError {
+      inner: e.into(),
+      status_code: None,
+    }
+  }
+}
+
+impl From<serde_json::Error> for LemmyError {
+  fn from(e: serde_json::Error) -> Self {
+    LemmyError {
+      inner: e.into(),
+      status_code: None,
+    }
+  }
+}
+
+impl From<strum::ParseError> for LemmyError {
+  fn from(e: strum::ParseError) -> Self {
+    LemmyError {
+      inner: e.into(),
+      status_code: None,
+    }
+  }
+}
+
+impl From<url::ParseError> for LemmyError {
+  fn from(e: url::ParseError) -> Self {
+    LemmyError {
+      inner: e.into(),
+      status_code: None,
+    }
+  }
+}
+
+impl From<FromStrError> for LemmyError {
+  fn from(e: FromStrError) -> Self {
+    LemmyError {
+      inner: e.into(),
+      status_code: None,
+    }
+  }
+}
+
+impl From<DomainError> for LemmyError {
+  fn from(e: DomainError) -> Self {
+    LemmyError {
+      inner: e.into(),
+      status_code: None,
+    }
+  }
+}
+
+impl From<ErrorStack> for LemmyError {
+  fn from(e: ErrorStack) -> Self {
+    LemmyError {
+      inner: e.into(),
+      status_code: None,
+    }
+  }
+}
+
+impl From<InvalidHeaderName> for LemmyError {
+  fn from(e: InvalidHeaderName) -> Self {
+    LemmyError {
+      inner: e.into(),
+      status_code: None,
+    }
+  }
+}
+
+impl From<InvalidHeaderValue> for LemmyError {
+  fn from(e: InvalidHeaderValue) -> Self {
+    LemmyError {
+      inner: e.into(),
+      status_code: None,
+    }
+  }
+}
+
+impl From<PrepareVerifyError> for LemmyError {
+  fn from(e: PrepareVerifyError) -> Self {
+    LemmyError {
+      inner: e.into(),
+      status_code: None,
+    }
+  }
+}
+
+impl From<PrepareSignError> for LemmyError {
+  fn from(e: PrepareSignError) -> Self {
+    LemmyError {
+      inner: e.into(),
+      status_code: None,
+    }
+  }
+}
+
+impl From<SignError> for LemmyError {
+  fn from(e: SignError) -> Self {
+    LemmyError {
+      inner: e.into(),
+      status_code: None,
+    }
+  }
+}
+
+impl From<DecodeError> for LemmyError {
+  fn from(e: DecodeError) -> Self {
+    LemmyError {
+      inner: e.into(),
+      status_code: None,
+    }
+  }
+}
+
+impl From<ParseIntError> for LemmyError {
+  fn from(e: ParseIntError) -> Self {
+    LemmyError {
+      inner: e.into(),
+      status_code: None,
+    }
+  }
+}
+
+impl From<std::io::Error> for LemmyError {
+  fn from(e: std::io::Error) -> Self {
+    LemmyError {
+      inner: e.into(),
+      status_code: None,
+    }
+  }
+}
+
```

Comment:
@asonix we're trying to add explicit error types for all of the above, but there's one activitystreams one we can't figure out:

```
error[E0277]: `?` couldn't convert the error to `LemmyError`
   --> lemmy_apub/src/activity_queue.rs:229:42
    |
229 |   let activity = activity.into_any_base()?;
    |                                          ^ the trait `From<<T as Extends<Kind>>::Error>` is not implemented for `LemmyError`
    |
    = note: the question mark operation (`?`) implicitly performs a conversion on the error value using the `From` trait
    = note: required by `std::convert::From::from`
```

Do you know of any way we could add that explicitly?